### PR TITLE
Report billing events

### PR DESCRIPTION
**Pull request by paolodamico**
Adds `billing subscription activated`, `billing subscription paid` & `billing card validated` events to be reported to PostHog.﻿

### CODE REVIEW

File: multi_tenancy/tasks.py [hunk 1]
```diff
@@ -50,22 +47,47 @@ def _compute_daily_usage_for_organization(
...
     )
 
     if not success:
         raise self.retry()
+
+
+@app.task(ignore_result=True, max_retries=3)
+def report_invoice_payment_succeeded(organization_id: Organization, initial: bool) -> None:
+
+    organization = Organization.objects.get(id=organization_id)
```
Comment (kpthatsme) [inline multi_tenancy/tasks.py:68]: It looks like we're passing in an Organization object here and then passing that in directly as the ID – do we need to pass in the ID field instead? Perhaps that's automatically parsed out

File: multi_tenancy/tasks.py [hunk 2]
```diff
@@ -50,22 +47,47 @@ def _compute_daily_usage_for_organization(
...
+    payload = {
+        "plan_key": organization.billing.get_plan_key(only_active=False),
+        "billing_period_ends": organization.billing.billing_period_ends,
```
Comment (kpthatsme) [inline multi_tenancy/tasks.py:71]: Thoughts on capturing Org ID or name in this payload?

I feel like a lot of the billing stuff we'll wanna take a look @ the org level as well

File: multi_tenancy/tasks.py (hunk 1 above)
Comment (kpthatsme) [reply multi_tenancy/tasks.py:68]: Ah I actually see that you pass in the ID from where you call this – so is `Organization` the right type for this param? Or should we use UUID?

**Review by kpthatsme — APPROVED** [review]
Left a few comments but generally looks good! This is going to be great to get in!

File: multi_tenancy/tasks.py (hunk 1 above)
Comment (paolodamico) [reply multi_tenancy/tasks.py:68]: Great catch! We don't run `mypy` on this repo, so it fell through the cracks. While it's indeed a UUID (actually a similar ID that doesn't quite conform to a UUID4 or similar standard created by @Twixes, optimized for database operations), as it's serialized through celery, the value will be passed as a string. Updated.

For context, when doing async tasks, it's best to pass down only primitive types only as I believe we don't support complex pickling (and it's more bug-prone anyways).

File: multi_tenancy/tasks.py (hunk 2 above)
Comment (paolodamico) [reply multi_tenancy/tasks.py:71]: Good idea! We're not actually doing this anywhere, but we should start. Addressed.

**Pull request merged**